web/src/beta/components/Popover/hooks.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  autoUpdate,
+  flip,
+  offset,
+  shift,
+  useClick,
+  useDismiss,
+  useFloating,
+  useInteractions,
+  useRole,
+} from "@floating-ui/react";
+import * as React from "react";
+
+import { PopoverOptions } from "./types";
+
+export default function usePopover({
+  initialOpen = false,
+  placement = "bottom",
+  modal,
+  open: controlledOpen,
+  onOpenChange: setControlledOpen,
+}: PopoverOptions = {}) {
+  const [uncontrolledOpen, setUncontrolledOpen] = React.useState(initialOpen);
+
+  const open = controlledOpen ?? uncontrolledOpen;
+  const setOpen = setControlledOpen ?? setUncontrolledOpen;
+
+  const data = useFloating({
+    placement,
+    open,
+    onOpenChange: setOpen,
+    whileElementsMounted: autoUpdate,
+    middleware: [
+      offset(4),
+      flip({
+        crossAxis: placement.includes("-"),
+        fallbackAxisSideDirection: "end",
+        padding: 4,
```

Comment:
Can we make the padding a prop as well? 🙏🏻 Unless you checked all current designs and the popover is the same distance away from its parent in all current cases.